### ISSUE TITLE
Windows Install

### ISSUE THREAD

**Issue by gismcieri** [openvenues/libpostal#64]
Before I try to reinvent the wheel. Anyone have a tips or procedure to install this on windows.
I would like to use it with windows 7 and python and possibly postgres.

Thanks,
Matt

**Comment by thatdatabaseguy** [comment]
Libpostal supports OSX and Linux for the time being as I have no Windows machines to test on, but patches are welcome.

As a workaround, Paul Ramsey wrote a Postgres binding [here](https://github.com/pramsey/pgsql-postal), so it would be possible to run a Postgres instance on Linux and connect from a Windows machine if that's kosher in your environment.

**Comment by gismcieri** [comment]
thanks for the response postgres on linux is possible and if i can get it to work on ubuntu i will give see what i can do about making it work on windows.

i have tried the install 2X and i keep getting
ERR   Error loading transliteration module
   at libpostal_setup (libpostal.c:1057) errno: No such file or directory

everything seems to install properly with no errors

Any thoughts?

Thanks!

**Comment by thatdatabaseguy** [comment]
Ok, so this is on Ubuntu?

If so, that error most likely means the data directory specified during configure didn't have enough space and the data files failed to download. I pushed a commit to master recently that will at least make the build fail if this happens. 

Try running ```du -h``` on whatever directory you used during configure. It should show about 2.2G being used (if running on a partition or VM, make sure there's enough disk space and also memory). If it's less than that, something didn't download and the simplest course of action is to delete the data directory and the libpostal checkout and install it from scratch. Let me know if that doesn't work.

**Comment by gismcieri** [comment]
I was reading one of the other issues and i think it may be a 32bit issue. I may have to bite the bullet and upgrade to 64bit ubuntu.  

I have more then enough space in my home directory and i use the git clone https://github.com/openvenues/libpostal command are you saying i should try from source.

Matt

**Comment by thatdatabaseguy** [comment]
This isn't 32-bit related (it's saying a file doesn't exist). libpostal is fully 32-bit compatible at this point.

I'm unclear on what try from source means. After git clone, did you run ```bootstrap.sh```, ```configure```, ```make``` and ```make install```? libpostal has to be installed before the Postgres extension can be built.

**Comment by gismcieri** [comment]
yes i i followed the setup steps in order and i understand that libpostal needs to be installed first.

The only thing different i do is make is requiring me to use sudo or it fails.  I will try it again tomorrow in case i missed something.

Thanks,
Matt

**Comment by gismcieri** [comment]
I got it. I actually upgraded my machine from 14.04 to 16.04 and when i went through the install everything worked fine.  I am not sure what happened before.

Tried it standalone with the command line and python. My next step is to try it with postgres. 

Thanks,
Matt

**Comment by janvidpa** [comment]
Hi @thatdatabaseguy 
When trying to run libpostal on ubuntu I get the following error:
ERR   language_classifier not loaded, run libpostal_setup_language_classifier()
   at classify_languages (language_classifier.c:63) errno: No such file or directory

I have followed the setup instructions. What am I missing here?

**Comment by thatdatabaseguy** [comment]
@gismcieri hooray! Re: sudo, sometimes if ```sudo make``` or ```sudo make install``` is run from a clean source checkout (or after something like ```make clean```), the compiled object files (*.o) can have root permissions and the build can fail on a sudoless ```make```.

**Comment by thatdatabaseguy** [comment]
@janvidpa that error means something didn't download. Quoting my advice above: try running ```du -h``` on whatever directory you passed in as --datadir during configure and make sure it shows 2.2G used. If it doesn't, the easiest course of action is to delete that directory and run ```make``` again (which will re-download the necessary files).

**Comment by janvidpa** [comment]
yes that was it. @thatdatabaseguy thanks! :)
